### PR TITLE
trezor-agent and dependencies

### PR DESCRIPTION
**Pull request by ananteris**

### CODE REVIEW

File: srcpkgs/python-ECPy/template [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Template file for 'python-ecpy'
+pkgname=python-ECPy
+version=0.8.1
+revision=1
+noarcha=yes
```
Comment (Vaelatern) [inline srcpkgs/python-ECPy/template]: Oh?

File: srcpkgs/python-ed25519/template [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Template file for 'python-ed25519'
+pkgname=python-ed25519
+version=1.4
+revision=1
+only_for_archs="i686 x86_64 x86_64-musl"
```
Comment (Vaelatern) [inline srcpkgs/python-ed25519/template]: Why the enumeration? Rather not enumerate but rather say "Doesn't work here but might be fixable".
Unless it contains x86 assembler or something.

File: srcpkgs/python-ed25519/template [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+wrksrc="ed25519-${version}"
+build_style=python2-module
+hostmakedepends="python-setuptools"
+makedepends="python-devel"
+depends="python"
+pycompile_module="ed25519.py"
```
Comment (Vaelatern) [inline srcpkgs/python-ed25519/template]: `="ed25519"`

File: srcpkgs/python-hidapi/template [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Template file for 'python-hidapi'
+pkgname=python-hidapi
+version=0.7.99.post15
+revision=1
+only_for_archs="i686 x86_64 x86_64-musl"
```
Comment (Vaelatern) [inline srcpkgs/python-hidapi/template]: Again, enumeration why?

File: srcpkgs/python-hidapi/template [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+wrksrc="hidapi-${version}"
+build_style=python2-module
+hostmakedepends="python-setuptools"
+makedepends="python-Cython libusb-devel python-devel eudev-libudev eudev-libudev-devel"
+depends="python"
+pycompile_module="hidapi.py"
```
Comment (Vaelatern) [inline srcpkgs/python-hidapi/template]: Doesn't belong I don't think. Installs egg info and a `.so`. @asergi ?

File: srcpkgs/python-ed25519/template (hunk 2 above)
Comment (ananteris) [inline srcpkgs/python-ed25519/template]: You mean why the arch restrictions? It is installing .so objects that only run on x86

File: srcpkgs/python-hidapi/template (hunk 4 above)
Comment (ananteris) [inline srcpkgs/python-hidapi/template]: Because it's installing compiled .so objects.

File: srcpkgs/python-ed25519/template (hunk 3 above)
Comment (ananteris) [inline srcpkgs/python-ed25519/template]: Are you sure this is correct? there are 297 packages that contain pycompile_module="something.py"

File: srcpkgs/python-hidapi/template (hunk 5 above)
Comment (ananteris) [inline srcpkgs/python-hidapi/template]: What does not belong?

File: srcpkgs/python-ECPy/template (hunk 1 above)
Comment (ananteris) [inline srcpkgs/python-ECPy/template]: I fixed the noarcha typo

File: srcpkgs/python-ed25519/template (hunk 3 above)
Comment (Vaelatern) [inline srcpkgs/python-ed25519/template]: Please also look at the list of files they install. Compare that with what you are installing. Rinse and repeat.

File: srcpkgs/python-hidapi/template (hunk 5 above)
Comment (Vaelatern) [inline srcpkgs/python-hidapi/template]: The line, at all.
You can't ask python to compile what is a binary and not python.

File: srcpkgs/python-hidapi/template (hunk 4 above)
Comment (ananteris) [inline srcpkgs/python-hidapi/template]: I've included a broken= for the build errors on the other architectures.

File: srcpkgs/python-hidapi/template (hunk 5 above)
Comment (ananteris) [inline srcpkgs/python-hidapi/template]: I've removed the pycompile_module on both hidapi and ed25519

File: srcpkgs/python-ed25519/template (hunk 2 above)
Comment (ananteris) [inline srcpkgs/python-ed25519/template]: Added broken= link for build failures on other archs

File: srcpkgs/python-ed25519/template (hunk 3 above)
Comment (ananteris) [inline srcpkgs/python-ed25519/template]: pycompile_module line removed.

**Comment by clehner** [comment]
@ananteris I successfully built and ran trezor-agent & deps from this branch on armv7l-musl (rpi3). Could the `only_for_archs` lines be removed for python-pycrypto, python-trezor, and trezor-agent, or else add `armv7l-musl` and others on which it might work?

File: srcpkgs/python-hidapi/template (hunk 4 above)
Comment (Vaelatern) [inline srcpkgs/python-hidapi/template]: Precompiled, provided by upstream, `.so` objects? As in, in the tarball?

And they link against `-musl`?

**Comment by ananteris** [comment]
Removed the remaining only_for_archs, should be good to go.

**Comment by clehner** [comment]
On Tue, 10 Jan 2017 15:15:40 -0800
ananteris <notifications@github.com> wrote:

> Removed the remaining only_for_archs, should be good to go.

Cool. trezor-agent too?

**Comment by ananteris** [comment]
Any update on this? Has been stale for almost a month.

**Comment by Vaelatern** [comment]
It's just a lot to review.

**Comment by ananteris** [comment]
If you have any outstanding questions about any of the packages let me know. Me and @clehner have gone through these and ran them. Everything is in working order.

**Comment by ananteris** [comment]
@Vaelatern There any update on this?

File: srcpkgs/python-ed25519/template [hunk 6]
```diff
@@ -0,0 +1,23 @@
+# Template file for 'python-ed25519'
+pkgname=python-ed25519
+version=1.4
+revision=2
```
Comment (Vaelatern) [inline srcpkgs/python-ed25519/template]: revision=1

File: srcpkgs/python-ed25519/template [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+license="Public Domain"
+distfiles="${PYPI_SITE}/e/ed25519/ed25519-${version}.tar.gz"
+checksum=2991b94e1883d1313c956a1e3ced27b8a2fdae23ac40c0d9d0b103d5a70d1d2a
+
+case "${XBPS_TARGET_MACHINE}" in
+	*-aarch64)
+		broken="https://travis-ci.org/voidlinux/void-packages/jobs/189610266" ;;
+	*-aarch64-musl)
+		broken="https://travis-ci.org/voidlinux/void-packages/builds/189610262" ;;
+esac
```
Comment (Vaelatern) [inline srcpkgs/python-ed25519/template]: Still broken?

File: srcpkgs/python-hidapi/template [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+case "${XBPS_TARGET_MACHINE}" in
+	*-aarch64)
+			broken="https://travis-ci.org/voidlinux/void-packages/jobs/189538727" ;;
+	*-aarch64-musl)
+			broken="https://travis-ci.org/voidlinux/void-packages/jobs/189538731" ;;
+	*-armv7hf)
+			broken="https://travis-ci.org/voidlinux/void-packages/jobs/189538728" ;;
+	*-armv7hf-musl)
+			broken="https://travis-ci.org/voidlinux/void-packages/jobs/189538730" ;;
```
Comment (Vaelatern) [inline srcpkgs/python-hidapi/template]: hostmakedepends should have `python-Cython`

File: srcpkgs/python-ledgerblue/template [hunk 9]
```diff
@@ -0,0 +1,15 @@
+# Template file for 'python-ledgerblue'
+pkgname=python-ledgerblue
+version=0.1.8
+revision=1
+wrksrc="ledgerblue-${version}"
+build_style=python2-module
+hostmakedepends="python-setuptools"
+depends="python python-ECPy python-Pillow python-future python-pycrypto python-Pillow>=3.4.0"
```
Comment (Vaelatern) [inline srcpkgs/python-ledgerblue/template]: python-Pillow appears many times.

File: srcpkgs/python-ed25519/template (hunk 7 above)
Comment (ananteris) [inline srcpkgs/python-ed25519/template]: Not still broken, removed. Compiles on my machine™

File: srcpkgs/python-ed25519/template (hunk 6 above)
Comment (ananteris) [inline srcpkgs/python-ed25519/template]: fixed

File: srcpkgs/python-ledgerblue/template (hunk 9 above)
Comment (ananteris) [inline srcpkgs/python-ledgerblue/template]: Good catch, sorry, removed.

File: srcpkgs/python-hidapi/template (hunk 8 above)
Comment (ananteris) [inline srcpkgs/python-hidapi/template]: fixed

Comment (Vaelatern) [inline srcpkgs/python-hidapi/template]: And then remove these broken things. It's no longer true.

File: srcpkgs/python-pycrypto/template [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+hostmakedepends="python-setuptools"
+makedepends="python-devel"
+depends="python"
+pycompile_module="pycrypto.py"
+short_desc="Cryptographic modules for Python"
+maintainer="ananteris <ananteris@protonmail.ch>"
+homepage="https://www.dlitz.net/software/pycrypto/"
+license="custom"
+distfiles="${PYPI_SITE}/p/pycrypto/pycrypto-${version}.tar.gz"
+checksum=f2ce1e989b272cfcb677616763e0a2e7ec659effa67a88aa92b3a65528f60a3c
```
Comment (Vaelatern) [inline srcpkgs/python-pycrypto/template]: `nocross=yes` unless you have a fix...

**Comment by asergi** [comment]
Just at a glance:
- `xbps-query -Rs python-crypto`
- all `pycompile_module` are wrong
- python3 flavor missing

File: srcpkgs/python-hidapi/template (hunk 8 above)
Comment (ananteris) [inline srcpkgs/python-hidapi/template]: You're thinking of the python-ed25519 package.. python-hidapi is still broken for these archs

File: srcpkgs/python-pycrypto/template (hunk 10 above)
Comment (ananteris) [inline srcpkgs/python-pycrypto/template]: Removed this entire package and updated python-ledgerblue to the correct (existing) dep

**Comment by ananteris** [comment]
@asergi would this cause problems if it was installed on a system without the python3 flavors added?

**Comment by Vaelatern** [comment]
No, but python3 users would be unable to use these packages.

**Comment by asergi** [comment]
No progress for months, closing. Feel free to resubmit this when you're ready.

**Pull request closed without merging**